cli/azd/pkg/pipeline/pipeline_manager.go

```diff
@@ -289,19 +288,23 @@ func (pm *PipelineManager) Configure(ctx context.Context) (result *PipelineConfi
 	displayMsg = fmt.Sprintf("Configuring repository %s to use credentials for %s", repoSlug, spConfig.applicationName)
 	pm.console.ShowSpinner(ctx, displayMsg, input.Step)
 
+	subscriptionId := pm.env.GetSubscriptionId()
+	credentials := &azcli.AzureCredentials{
+		ClientId:       servicePrincipal.AppId,
+		TenantId:       *servicePrincipal.AppOwnerOrganizationId,
+		SubscriptionId: subscriptionId,
+	}
+
 	// Get the requested credential options from the CI provider
-	credentialOptions := pm.ciProvider.credentialOptions(
+	credentialOptions, err := pm.ciProvider.credentialOptions(
 		ctx,
 		gitRepoInfo,
 		infra.Options,
 		PipelineAuthType(pm.args.PipelineAuthTypeName),
+		credentials,
 	)
-
-	subscriptionId := pm.env.GetSubscriptionId()
-	credentials := &azcli.AzureCredentials{
-		ClientId:       servicePrincipal.AppId,
-		TenantId:       *servicePrincipal.AppOwnerOrganizationId,
-		SubscriptionId: subscriptionId,
+	if err != nil {
+		return result, fmt.Errorf("failed to get credential options: %w", err)
```

Comment:
I suspect we need to call `pm.console.StopSpinner` here as well?

I feel like the `Show/Stop` pattern we have interacts poorly with the fact that go code usually has multiple return points.

I almost wonder if we should be building a helper that takes a func that returns a result and error and it handles the start/stop for your or having `StartSpinner` return some object that you can `defer` to stop the spinner (perhaps leveraging a named return to check the error) would make it less likely for us to run into issues here?

Maybe something we can discuss with the larger team at the engineering sync.

Reply:
We have a pattern someplace to handle this. All credits to @weikanglim  :)

The strategy is to define and re-use the error ( `var err error` ) in a defer statement, like:

![image](https://github.com/Azure/azure-dev/assets/24213737/cbfc9baf-1cee-4e26-8ebc-251862d0c7b2)

Then, no matter a what point of the method we return an error, the spinner will be stopped and the cal to `input.GetStepResultFormat(err)` will display either a pass green check or a fail.